pkg/util/maps/maps.go

```diff
@@ -53,6 +53,31 @@ func Merge[K comparable, V any, S ~map[K]V](a, b S, resolveConflict func(a, b V)
 	return ret
 }
 
+func Intersect[K comparable, V any, M ~map[K]V](a, b M, val func(a, b V) V) M {
```

Comment:
Uhm. It seems that `Merge()` and `Intersect()` are much similar. Maybe we can consolidate these methods into one. @trasc WDYT?

Reply:
done